src/test/java/io/candydoc/ddd/extract_ddd_concepts/ExtractDDDConceptsUseCaseTest.java

```diff
@@ -307,6 +309,95 @@ void find_interaction_between_two_different_concepts() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.valid_bounded_contexts")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            SharedKernelFound.builder()
+                .name("shared_kernel_one")
+                .description("description of shared kernel")
+                .packageName("io.candydoc.sample.valid_bounded_contexts.shared_kernel")
+                .build());
+  }
+
+  @Test
+  void find_core_concepts_inside_shared_kernel() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.valid_bounded_contexts.shared_kernel")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            CoreConceptFound.builder()
+                .simpleName("name of core concept 1 of shared kernel 1")
+                .description("description of core concept 1 of shared kernel 1")
+                .canonicalName(
+                    "io.candydoc.sample.valid_bounded_contexts.shared_kernel.sub_package.CoreConcept1")
+                .packageName("io.candydoc.sample.valid_bounded_contexts.shared_kernel.sub_package")
+                .boundedContext("io.candydoc.sample.valid_bounded_contexts.shared_kernel")
+                .build());
+  }
+
+  @Test
+  void inner_bounded_context_are_forbidden() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.wrong_bounded_context")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            ConceptRuleViolated.builder()
+                .conceptName(
+                    "io.candydoc.sample.wrong_bounded_contexts.bounded_context.sub_package.bounded_context_two")
+                .reason(
+                    "bounded_context_two"
+                        + " shouldn't be in another bounded context/shared kernel.")
+                .build());
+  }
+
+  @Test
+  void inner_shared_kernel_are_forbidden() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.wrong_bounded_contexts")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            SharedKernelFound.builder()
+                .name("shared_kernel_two")
+                .description("description of shared kernel 2")
```

Comment:
for shared kernels, you'll have more interaction to check like Aggregates, Domain Commands, Domain Events, basically everything that's not a ValueObject nor a CoreConcept is forbidden